### ISSUE TITLE
Potential PRs

### ISSUE THREAD

**Issue by Nicholas-Mitchell** [daavoo/pyntcloud#192]
Hi @daavoo 

I have been using this magnificent package and found a couple of things that I had to do manually and alter, which could easily be integrated into the package. They aren't big things, but I'd be happy to discuss needs/requirements/objections before I were to go ahead and submit pull requests.

First one: **read binary files with `.bin` extension**
I think this might be stepping on the toes of a few other `FROM` functions already implemented, but also that this is perhaps going to be a very common task for users. This may especially be the case with the rise of LiDAR usage in the context of autonomous driving. Any hints on making this compliant with the existing API would be appreciated.

PR suggestion # 2: **adding default point size to PyntCloud.plot()**
The scenes rendered (at least in Jupyter notebooks) always begin with a default value of 0.00, which must be manually adjusted. Untli that is done, no points are rendered, just the background.
I have seen examples suggesting this used to be possible. was it removed or didn't it get implemented? If the former, why?

Any thoughts, suggestions?

Thanks

**Comment by daavoo** [comment]
Hola @Nicholas-Mitchell . Really glad to hear that you find pyntcloud usefull :) and even more glad to hear that you contribute 🎉 

---
I'm a little confused because I'm not really familiar with the `.bin` format. 

The only `.bin` extension that I have ever encountered using point clouds is the [CloudCompare custom format](http://www.cloudcompare.org/forum/viewtopic.php?t=1864); is this what you are refering to?? Or are you refering to some generic binary format??

If you could clarify what format are you refering to, it would help me give the right indications.

Regardless of that, If you would like to add a new format (let's call if .foo) to `io` you should:

- Add a new python file (`foo.py`) to `pyntcloud/io`
- Implement inside that file read_foo() and/or write_foo(). You can/should use as reference the existing functions for other formats.  Basically, you should load the data into a pandas DataFrame in `read_foo` and accept a pandas DataFrame as argument in `write_foo`.
- Add a new entry to the `FROM` and/or `TO` dictionaries in `pyntcloud/io/__init__.py`
- Add some tests to tests/unit/io and tests/integration/io

I think that if you navigate throught the existing io functions you should get a better idea of what to do.

---

It is true that the initial point size is 0.0 but in the point clouds that I've tried this doesn't mean that only the background is rendered; the points are really small, but visible. It didn't get implemented because of that but I think that this might be a good addition.

You could either accept an initial point_size as argument to PyntCloud.plot() or you could compute a default initial value higher that 0, similar to how the min and max values are computed in:

https://github.com/daavoo/pyntcloud/blob/master/pyntcloud/core_class.py#L746

I think that this line is the same where you would set the initial default value using the `value` argument, according to how the FloatSlider is documented in:

https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20List.html#FloatSlider

**Comment by Nicholas-Mitchell** [comment]
@daavoo - Great, thanks for the tips!

The binary files could be anything really. I haven't worked with the CloudCompare tools, but have used binary files a fair bit, coming directly from LiDAR sensors. I started using PyntCloud recently after looking at data from the KITTI dataset - autonomous driving data recorded from a 64 channel Velodyne LiDAR.
As an aside: people also quite often use `.bin` for Protobuf files... but if people are using those, they hopefully also know how to deal with them :wink:.

The initial point size is then perhaps dependent on things such as screen resolution - On a 1920x1080 screen I cannot see anything :disappointed:  - even if I zoom in.
However, if that works fine for you right now, maybe it does for others. I will just leave the default value of zero, but allow the user to pass a value into `init_point_size`. I think it is intuitive enough for beginners (like me last week!) to then use the actual slider to increase the point size.
That does indeed get passed to FloatSlider(value=init_point_size, ...) as you say.

I have already implemented some things, so will submit a pull request sometime soon :+1:
